expression/distsql_builtin.go

```diff
@@ -684,10 +684,15 @@ func convertFloat(val []byte, f32 bool) (*Constant, error) {
 
 func convertDecimal(val []byte) (*Constant, error) {
 	_, dec, err := codec.DecodeDecimal(val)
+	var d types.Datum
```

Comment:
DRY. We should abstract a function, it's duplicate with  util/codec/codec.go at line 339